core/src/lib/plugin_registry.h

```diff
@@ -0,0 +1,75 @@
+/*
```

Comment:
Maybe better call it "backend registry" to avoid confusion with fd, sd and dir plugins?

Reply:
renamed it to ImplementationFactory